bundles/org.openhab.binding.icalendar/src/main/java/org/openhab/binding/icalendar/internal/logic/CommandTag.java

```diff
@@ -0,0 +1,167 @@
+
+    private @Nullable String authorizationCode;
+    private String fullTag;
+    private boolean isValid = false;
+    private @Nullable String itemName;
+    private final Logger logger = LoggerFactory.getLogger(CommandTag.class);
+    private @Nullable CommandTagType tagType;
+    private @Nullable String targetState;
+
+    public CommandTag(String line) {
+        fullTag = line;
+        if (!fullTag.contains(":")) {
+            logger.trace("Input line \"{}\" => No \":\" delimiters!", fullTag);
+            return;
+        }
+        String[] fields = fullTag.split(":");
+        if (fields.length < 3) {
+            logger.trace("Input line \"{}\" => Not enough fields!", fullTag);
+            return;
+        }
+        try {
+            tagType = CommandTagType.valueOf(fields[0]);
+        } catch (IllegalArgumentException e) {
+            logger.trace("Input line \"{}\" => Bad tag prefix!", fullTag);
+            return;
+        }
+        String newItemName = fields[1].trim();
+        if (newItemName.isEmpty()) {
+            logger.trace("Input line \"{}\" => Item name empty!", fullTag);
+            return;
+        } else {
+            itemName = newItemName;
+        }
+        String newTargetState = fields[2].trim();
+        if (newTargetState.isEmpty()) {
+            logger.trace("Input line \"{}\" => Target State empty!", fullTag);
+            return;
+        } else {
+            targetState = newTargetState;
+        }
+        isValid = true;
+        if (fields.length > 3) {
+            authorizationCode = fields[3].trim();
+        }
+    }
+
+    @Nullable
+    public static CommandTag createCommandTag(String line) {
+        if (CommandTagType.prefixValid(line)) {
+            CommandTag tag = new CommandTag(line.trim());
+            return tag.isValid ? tag : null;
+        } else {
+            return null;
+        }
+    }
+
+    public @Nullable Command getCommand() {
+        String currentTargetState = targetState;
+        if (currentTargetState == null || currentTargetState.isEmpty()) {
+            return null;
+        }
+
+        // string is in double quotes => force StringType
+        if (currentTargetState.startsWith("\"") && currentTargetState.endsWith("\"")) {
+            return new StringType(currentTargetState.replaceAll("\"", ""));
+        }
+
+        // string is in single quotes => ditto
+        if (currentTargetState.startsWith("'") && currentTargetState.endsWith("'")) {
+            return new StringType(currentTargetState.replaceAll("'", ""));
+        }
+
+        Command cmd;
+
+        // string ends with % => try PercentType
+        if (currentTargetState.endsWith("%")) {
+            if ((cmd = TypeParser.parseCommand(percentCommandType,
+                    currentTargetState.substring(0, currentTargetState.length() - 1))) != null) {
+                return cmd;
+            }
```

Comment:
Please don't do variable assignment inside your if condition. The only exception to this rule is reading from streams in a loop.

Reply:
OK.